src/app/projects/states/dashboard/directives/task-dashboard/directives/task-status-card/task-status-card.coffee

```diff
@@ -43,10 +43,24 @@ angular.module('doubtfire.projects.states.dashboard.directives.task-dashboard.di
           "This task is past the target due date, so you will need to request an extension to get feedback. Use confirm to request extensions and proceed with the submission.",
           () -> extendAndSubmit(trigger))
       else
-        $scope.task.triggerTransition(trigger, $scope.unitRole)
+        $scope.triggerStatusChange(trigger)
 
     # Allow upload of new files for an updated submission
     $scope.updateFilesInSubmission = ->
       taskService.presentTaskSubmissionModal($scope.task, $scope.task.status, true)
 
+    $scope.triggerStatusChange = (trigger) ->
+      if (trigger == 'not_started' || trigger == 'working_on_it' || trigger == 'need_help') && $scope.task.status == 'ready_to_mark' && !$scope.task.isOverdue()
+        ConfirmationModal.show(
+          "Do you want to change the status?",
+          "Changing the status will cancel all previous submissions.
+          Use confirm to change the status.",
+          () -> $scope.task.triggerTransition(trigger, $scope.unitRole))
+      else if (trigger == 'not_started' || trigger == 'working_on_it' || trigger == 'need_help') && $scope.task.status == 'ready_to_mark' && $scope.task.isOverdue()
+        WarningModal.show(
+          "Not allowed to perform this action",
+          "The deadline for this task has passed and you are not allowed to change status.
```

Comment:
"The deadline for this task has passed and you are not allowed to change *the* status."